// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: confirm_email.proto

package v1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on ConfirmEmail with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *ConfirmEmail) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ConfirmEmail with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in ConfirmEmailMultiError, or
// nil if none found.
func (m *ConfirmEmail) ValidateAll() error {
	return m.validate(true)
}

func (m *ConfirmEmail) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Email

	// no validation rules for UserName

	// no validation rules for ConfirmationCode

	// no validation rules for Token

	if len(errors) > 0 {
		return ConfirmEmailMultiError(errors)
	}
	return nil
}

// ConfirmEmailMultiError is an error wrapping multiple validation errors
// returned by ConfirmEmail.ValidateAll() if the designated constraints aren't met.
type ConfirmEmailMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ConfirmEmailMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ConfirmEmailMultiError) AllErrors() []error { return m }

// ConfirmEmailValidationError is the validation error returned by
// ConfirmEmail.Validate if the designated constraints aren't met.
type ConfirmEmailValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ConfirmEmailValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ConfirmEmailValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ConfirmEmailValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ConfirmEmailValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ConfirmEmailValidationError) ErrorName() string { return "ConfirmEmailValidationError" }

// Error satisfies the builtin error interface
func (e ConfirmEmailValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sConfirmEmail.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ConfirmEmailValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ConfirmEmailValidationError{}
